### PR TITLE
Fixed asynchronous event scheduling

### PR DESCRIPTION
**Pull request by TheBusyBiscuit**
# Description
This pull requests addresses some inconsistent or wrong behaviours within the way the `AsyncPlayerChatEvent` is handled.
This PR is a combined effort from #187 and #186.

@Narimm In `PluginManagerMock` (line 419) you can find a reference to the method from `BukkitSchedulerMock` that will be your part. Feel free to rename, refactor or restructure this to your liking, it is just a skeleton placeholder from my part for now.
Simply make a new pull request and select the `fix/async-events` branch as the target and I will merge the changes into this PR.

Also note that the compilation will fail until the newly added Spigot methods are added (see #187)

# Checklist
The following items should be checked before the pull request can be merged.
- [ ] Version number updated in `settings.gradle`.
- [ ] Unit tests added.
- [x] Code follows existing code format.

### CODE REVIEW

**Comment by Narimm** [comment]
Just a heads up #189 is ready to merge into this PR

**Review by Narimm — APPROVED** [review]

**Comment by seeseemelk** [comment]
Honestly, this is kinda hard to review as there a waaayyy too many `@NotNulls` there were introduced here. I am having trouble finding what the actual meat is of this PR. Could you perhaps create a PR with just the annotations so that I can merge and review this one much more easily?

**Comment by TheBusyBiscuit** [comment]
I can understand, took me some time to go through these too.
I went ahead and reverted some of Narimm's annotations for now to aid with the reviewing process, we can always re-add them later.

**Comment by seeseemelk** [comment]
I created #200 which copies just the annotations from the PR. When that one is merged, this one should be much easier to read.

File: src/main/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMock.java [hunk 1]
```diff
@@ -459,12 +483,16 @@ private void addSection(PluginCommand command, String name, Object value)
...
-				    ((List<?>) aliases).stream().map(Object::toString).collect(Collectors.toList()));
+						((List<?>)value).stream().map(Object::toString).collect(Collectors.toList()));
 			else
-				command.setAliases(Arrays.asList(value.toString()));
+				if(value != null)
+				{
+					command.setAliases(Collections.singletonList(value.toString()));
+				} else {
+					command.setAliases(Collections.emptyList());
+				}
```
Comment (seeseemelk) [inline src/main/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMock.java]: This if-statement is formatted really weirdly

**Review by seeseemelk — CHANGES_REQUESTED** [review]
Aside from this one code style issue, this seems good to me! Should I close #183 and #186 as this PR supersedes these (I think)?

File: src/main/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMock.java (hunk 1 above)
Comment (TheBusyBiscuit) [reply src/main/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMock.java]: Fixed.

**Comment by TheBusyBiscuit** [comment]
#183 and #186 don't really have much to do with this. This right here just fixes the way async events are called since that was somewhat broken.
Whereas the other two PRs deal with the scheduler which I know nothing about :eyes:

File: src/main/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMock.java [hunk 2]
```diff
@@ -484,15 +484,17 @@ private void addSection(PluginCommand command, String name, Object value)
...
-				} else {
-					command.setAliases(Collections.emptyList());
-				}
+						((List<?>) value).stream().map(Object::toString).collect(Collectors.toList()));
+			}
+			else if (value != null)
+			{
+				command.setAliases(Collections.singletonList(value.toString()));
+			}
+			else {
```
Comment (seeseemelk) [inline src/main/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMock.java]:
```suggestion
			else
			{
```

**Review by seeseemelk — CHANGES_REQUESTED** [review]
So close...

**Comment by seeseemelk** [comment]
> #183 and #186 don't really have much to do with this. This right here just fixes the way async events are called since that was somewhat broken.
> Whereas the other two PRs deal with the scheduler which I know nothing about :eyes:

Aight, then I'll leave those two open and revisit them at a later time

**Review by seeseemelk — APPROVED** [review]

**Pull request merged**